src/components/common/DropDown/DropDownSelector.tsx

```diff
@@ -0,0 +1,96 @@
+import React, {useState, useRef, useMemo} from 'react';
+import {
+  TouchableOpacity,
+  FlatList,
+  ScrollView,
+  Keyboard,
+  TouchableWithoutFeedback,
+} from 'react-native';
+import S from './DropDownSelector.style';
+
+type Props = {
+  value: string[];
+  options: string[];
+  onChange: (val: string) => void;
+  placeholder?: string;
+};
+
+const DropDownSelector = ({value, options, onChange, placeholder}: Props) => {
+  const [newOptionName, setNewOptionName] = useState<string>('');
+  const [isDropdownVisible, setIsDropdownVisible] = useState<boolean>(false);
+  const inputRef = useRef(null);
+
+  const handleInputChange = (input: string) => {
+    const trimmedInput = input.trim();
+    setNewOptionName(trimmedInput);
+    setIsDropdownVisible(trimmedInput.length > 0);
+  };
```

Comment:
```suggestion
    setNewOptionName(trimmedInput);
    setIsDropdownVisible(Boolean(trimmedInput));
  };
```

length > 0 이면 truly 라서 !! 연산자 혹은 Boolean 연산자를 사용할 수 있겠네요.

밴치마킹 돌려보니 이렇네요... ( DATA 배열 길이 :100,000,0 )

<img width="1377" alt="스크린샷 2024-10-06 오전 2 19 59" src="https://github.com/user-attachments/assets/cf781502-b16b-48e2-86cf-dc430a5671fa">

Reply:
이런 정보 매우 감사합니다..